runtime/context.go

```diff
@@ -165,6 +166,59 @@ func ServerMetadataFromContext(ctx context.Context) (md ServerMetadata, ok bool)
 	return
 }
 
+// ServerTransportStream implement grpc.ServerTransportStream
+type ServerTransportStream struct {
+	mu      sync.Mutex
+	header  metadata.MD
+	trailer metadata.MD
+}
+
+// Method returns the method for the stream.
+func (s *ServerTransportStream) Method() string {
+	return ""
+}
+
+// Header returns the header metadata of the stream.
+func (s *ServerTransportStream) Header() metadata.MD {
+	c := s.header.Copy()
```

Comment:
Doesn't this need to be mutex protected?
```suggestion
	s.mu.Lock()
	defer s.mu.Unlock()
	c := s.header.Copy()
```

Reply:
Maybe, but the original library does not use a mutex in its implementation
https://github.com/grpc/grpc-go/blob/master/internal/transport/transport.go#L386

Reply:
Interesting, seems pretty specific to their design though. This design definitely needs it.